docs/user/gcp/install_upi.md

```diff
@@ -98,6 +123,48 @@ del data["spec"]["privateZone"];
 open(path, "w").write(yaml.dump(data, default_flow_style=False))'
 ```
 
+```console
+spec:
+  baseDomain: example.com
+```
+
+### Update the cloud-provider manifest ([Shared VPC (XPN)][sharedvpc] only)
+If you are installing into a [Shared VPC (XPN)][sharedvpc],
+update the cloud provider configuration so it understands the network and subnetworks are in a different project (host project).
+Otherwise skip this step.
+
+```sh
+export HOST_PROJECT="example-shared-vpc"
+export HOST_PROJECT_NETWORK_NAME="example-network"
+export HOST_PROJECT_COMPUTE_SUBNET_NAME="example-worker-subnet"
+
+sed -i "s/    subnetwork-name.*/    network-project-id = ${HOST_PROJECT}\\n    network-name    = ${HOST_PROJECT_NETWORK_NAME}\\n    subnetwork-name = ${HOST_PROJECT_COMPUTE_SUBNET_NAME}/" manifests/cloud-provider-config.yaml
+```
+
+```console
+  config: |+
+    [global]
+    project-id      = example-project
+    regional        = true
+    multizone       = true
+    node-tags       = opensh-ptzzx-master
+    node-tags       = opensh-ptzzx-worker
+    node-instance-prefix = opensh-ptzzx
+    external-instance-groups-prefix = opensh-ptzzx
+    network-project-id = example-shared-vpc
+    network-name    = example-network
+    subnetwork-name = example-worker-subnet
+```
+
+### Enable external ingress (optional)
+If you are installing into a [Shared VPC (XPN)][sharedvpc],
+and you set `publish: Internal` in the `install-config.yaml` but really wanted `publish: External`
+then remove the `cluster-ingress-default-ingresscontroller` manifest to enable external ingress.
+
+```sh
+rm -f manifests/cluster-ingress-default-ingresscontroller.yaml
+```
```

Comment:
hmm, i don't think removing is a solution that the ingress team will like, this probably needs to update the object correctly.

Reply:
This file only exists when `publish: Internal`. It is not clear what it should look like, if at all, when `publish: External`

Reply:
Updated with f512a3c77d5f811401e7ecd675e4227bd8c694d7 to only modify the file from `Internal` to `External`